bylaws/008-psr-evolution.md

```diff
@@ -0,0 +1,69 @@
+# PSR evolution
+
+## Scope and objectives
+
+A PSR is often comprised of text and code, more specifically interfaces. Those interfaces are pieces of code that are released and tagged at a specific moment in time, but the PHP language doesn't stand still... it evolves over time.
+
+This means that those interfaces need to withstand that evolution, and sometimes they need to be updated, to leverage new language features that could help better enforce the behaviors exposed in the PSR itself.
+At the same time, a PSR cannot be changed after its release (only erratas allowed), to protect a package that declared compatibility from becoming de-facto not compatible anymore.
+
+This document defines a process to be followed in updating PSR interfaces, in a way that is not breaking in regard to behavior for end users, and with an appropriate upgrade path for the packages.
+
+## New releases
+
+A new minor release of a PHP-FIG package containing interfaces for a PSR MUST follow these rules:
+ * the new release MUST follow [Semantic Versioning](https://semver.org/) rules;
+ * the PSR behavior MUST NOT be altered;
+ * the packages that implement the interfaces or users that are consuming them MUST NOT suffer breaking changes;
+ * the PHP version constraint of the PHP-FIG package MAY be altered to require newer language features that would aid cross-compatibility;
+ * the PHP version constraint of the PHP-FIG package MUST NOT be altered to use newer language features that would create cross-compatibility issues.
+ 
+A new major release of a PHP-FIG package containing interfaces for a PSR MUST follow the same rules, with this exception:
+ * the new major version of the package MAY contain breaking changes if the implementing packages have a reasonable upgrade path, like the possibility of releasing a cross-compatible implementation with the previous releases;
```

Comment:
It is my concern that the language used, "have a reasonable upgrade path", to define an exception, is broad and vague and candidate for easy abuse. Perhaps it is safe to bind this to a clear example of what is explicitly NOT "reasonable".

I believe something specific can be formulated along the lines of:
> If the upgrade path causes userland to maintain multiple versions of their software side-by-side, only to support multiple versions of the same PSR, the upgrade path is too steep.

I'm unclear whether that's an improvement, but would like at least to have this considered by people more clever than myself.

Reply:
I agree with @johmanx10 here - "reasonable" is very subjective. Having examples of unreasonable and/or reasonable paths would help remove uncertainty and the possibility of unfair handling of different PSR evolutions.

I think the sentence Jan-Marten suggests is spot on.